### ISSUE TITLE
More search paths needed for SQLite adoption in v3.0.0

### ISSUE THREAD

**Issue by tedsmith** [tedsmith/quickhash#34]
Further to commit https://github.com/tedsmith/quickhash/commit/16e63916806db757fea83145a4097c729ea226ce @darealshinji has rightly pointed out that the cross platform nature of v3.0.0 will be limited to Debian based systems. 

Need to add the following search paths for SQLite: 

/usr/local/lib
/usr/lib
/usr/lib/i386-linux-gnu | /usr/lib/x86_64-linux-gnu
/usr/lib32 | /usr/lib64
/lib
/lib/i386-linux-gnu | /lib/x86_64-linux-gnu
/lib32 | /lib64

**Comment by darealshinji** [comment]
Does `SQLDBLibraryLoaderLinux.LibraryName := 'libsqlite3.so.0';` load the library too or does it have to be a full path?
I'm asking because in C/C++ `dlopen("libsqlite3.so.0", RTLD_LAZY);` would load the first encounter of `libsqlite3.so.0` in the library search path.

**Comment by tedsmith** [comment]
I did try it, initially, with just the filename as you suggest. In fact, when you create the component on the form, it defaults to that. But whenever I executed the program, I got database connection and creation errors. Yet if I enter the full path, it worked and I got no errors. 

So I think it should in theory work with just the default value. And that would, as you suggest, solve the Linux distribution problem. The problem is I can't easily develop if I have to change that value every time I switch to Linux compilation. Perhaps something to remember for beta release time - see how many users have problems with it just set to SQLDBLibraryLoaderLinux.LibraryName := 'libsqlite3.so.0';
